### PR TITLE
Update quarto-publish-example.yml

### PR DESCRIPTION
**Pull request by hamelsmu**
Add additional notes that I think might be helpful for people.

### CODE REVIEW

**Comment by hamelsmu** [comment]
When I was setting up Actions recently, I had to find these things, so I propose we add them here? 
 
@mcanouil @cderv

File: examples/quarto-publish-example.yml [hunk 1]
```diff
@@ -19,7 +24,13 @@ jobs:
...
           # version: SPECIFIC-QUARTO-VERSION-HERE
       
-      # add software dependencies here
+      # add software dependencies here, for example R or Python, and any libraries
+      
+      # - name: setup Python
+      #   uses: actions/setup-python@v3
+
+      # - name: setup R
+      #   uses: r-lib/actions/setup-r@v2
```
Comment (mcanouil) [inline examples/quarto-publish-example.yml:33]:
```suggestion
      # From https://github.com/actions/setup-python
      # - name: Setup Python
      #   uses: actions/setup-python@v3

      # From https://github.com/r-lib/actions/tree/v2-branch/setup-r
      # - name: Setup R
      #   uses: r-lib/actions/setup-r@v2

      # From https://github.com/julia-actions/setup-julia
      # - name: Setup Julia
      #   uses: julia-actions/setup-julia@v1
```

**Review by mcanouil — APPROVED** [review]
I added/suggested Julia action in addition to R since it is one of the main languages advertised in Quarto.
I also added the source of the actions for the documentation of those.

Seems to be a good addition since most users will have to add any of those actions to have computations (which is basically what offers Quarto).

**Review by cderv — COMMENTED** [review]
I am fine with adding them. However, let me explain why they are not currently: 

Uncommenting these lines is not enough to make it work, as you need to configure the environment too to install the packages required by each of those language. 
Like installing Python is not enough is you don't have jupyter install, and other used deps in your project. 

To avoid maintaining a long file with all the cases possible, we documented the dependencies with links and example in the doc page (https://github.com/quarto-dev/quarto-actions/blob/main/examples/example-03-dependencies.md
), and mentioned in this YAML example only the reminder to add dependencies. 

So I see two steps to improve this 

* Maybe add the link in this YAML workflow to https://github.com/quarto-dev/quarto-actions/blob/main/examples/example-03-dependencies.md

* Add a workflow example for each language. Complete one that works out of the box. 

Does that make sense ?

**Comment by hamelsmu** [comment]
@cderv I like your idea much better It would result in less clutter also

Let me close this PR

**Comment by hamelsmu** [comment]
In addition to language what about publishing target (netlify, GitHub pages etc?  How do you think that is best handled for complete working examples?

What we would have is Lang x Pub target so lots of combos 

Do you think the pub target should be commented ?

**Comment by hamelsmu** [comment]
Great idea about website tool!  maybe we have a small app in shiny live that helps people construct a yaml?

**Comment by cderv** [comment]
> maybe we have a small app in shiny live that helps people construct a yaml?

Yes exactly ! We need to make this small project comes live - timing constaint as often. Cool project though for anyone wanting to learn shinylive !

**Comment by hamelsmu** [comment]
Here is a basic [shiny app](https://shinylive.io/py/app/#code=NobwRAdghgtgpmAXGKAHVA6VBPMAaMAYwHsIAXOcpMAMwCdiYACAZwAsBLCbJjmVYnTJMArhzxM6lACZw6EgILoAOhFVpUAfTFMAvKI5YoAcziaaHAB5xpAClVNHBjGwBM9sAEURUIcSYA4hxkABIiAEZMCoRkHKQsTCFwADaocspgAJR4Dk5iGDC+ANbSxADuEB4Z1RBOdUwAKpwJqCZwTNIcLKjJUNhwCXCWsD0DTMQ0TCT8yXAUknCEHGkJNIKiLFzGTMBBoRFRMXEQLAC6tmxkZKgsiAD0d8bBbBEY03cAjj5+ALSyAG6fb5kYg-KBHeKZJggjpwHrEHhQJjeXwwzYUDBMXL1LG1HFMADKcHawDIbAG7SGI1mZwuVxu90ez1e7y+qNBAKB7LBEJOdzIUjgd0KXDuVJmAyhMI4yWSIhYAqg8zY5SY2GIIiYhR4hHlIJgHAAXu0yVsEkjwsEmDQREJyXQMNisWBqlkcnjHPleuqRGRNJtZOFfPYPfV8q0ICl-RxA8GnfjnFxUL7NHQoJ1iJpwr6QScPL0IMYMhIMgAZKCFnymYvxhNOToxWw4MmkXQZAAK2BbanwkjbYAASsWmAArETJDhQfsAKXHk4ymVrOOyS7DhiTKbTGazOfiHjIvlMZGHGQAInDkgj4ORGoe5jXQ3X6xxG8Y2P29mFIu22ixh5GyAnGhsH7AA5OYgOwYcPmTcJ+xRPwmHbCIF1XOoV0fJwMITcMKyjEVKjQvJDA1a4UzEDxsFgZJUMwxxF1DBiGNUWRJmmZMKE0KiYGSWwC2MCQDzoI9MkQJ02GkOg9GdV0AAN5O4miIGgeBECYAcZDkJgK2kJCIgndgnyMnFVFIMS8WTdhzLqcI0wgQhyVuLUoC4V0aidDhJiEo89H0DI3wyaz8QkqSAGo-JdSLVFUNI6ANFhNniNT4xIchKDIJyyjoYI4HjVpTEy7KKDcl01FDEKmHCmSorKkdiHCW5cmzGVpD+C8ESCxw6BEE4fjM0Rs3IEQfl6CgFSdBU4UazCfiYFS4DUgBhclCCKcZfQWAR0UEKC6I2AY1PBWJ4juBzFiKUiAAF-gAFlXVcAGJWGJJhLmuW4HieMkWUYLlfk5NlfiO44WH5QVhRciA7hYOYRFQa11nTaRjugZJxlQFG-xmubYAWwk5lEeGEJBVd5QOphAZBNrAUp0FgZOmGyDh67XHuqLSo8yZ+N851mxVHtOvqCqqpK2tZvmtSiSZ+HO27UmYac+m+UZuGfj50hroAZlXMpnkFnF1YgH5-jkRKIDUgByTWMEsC2mCegA1U3jkkCtTHGKSqRiJgTboM3xkmJFZf5n3ndIaF-DJiR5S2fGYCdugLYSX3-bs92WGwcgoEsaLsYlpgAEkTgPWUmFnZstLWKTieIVduvNphUGWXhi6gUuxwrugxZx1TC9b0vg-D2Q0ggWR7I4AY656tSm-hrgFTbtGfikqQvg4KRrwyjAyEsYRcQTUWyrqTymG53QIq7pBV2FiK3Lz3HJdh+Ghz2smnLoEaOHCO4ldBlXUGXizHWesiJdWNmHBuFsboYFcBgAADHbJ6TR2gDlDn7F2MJSgVEvOmJgtgT6RkIAMFgvhsBQh0vtR0R96jiwfn3BepdUHD00uPMY+98RvzUh-Cc39f7QyfsvNqI8x5LAGEAvausyT63qK0Vav5pE4grNgRAiAigQGCF3PadQlEqLisULB1Cnw6MQAYicx4tFOGMZYHirN2a52PlzN2PMMhjgnFAQK19JKVVvjVHEtDe5SzhmXOc7jX4KzUq4ycPJMb8Olj8SJUBroAEZu75yLgwtGs43GwmEZQURWMEz1wiSEpgPw1YMBHIsYQPx2gWxjoWJCRRjAAG5GnGAwPPA85BJwUFsJkC2h9ObQjvMIc+zoAKQQ8XtG+1Uah+J7njZC4QDJsAjkwcCgFPI8FsBQqQo85AMRwuEimwIORwBpqc6JIM7iwRWeIhMki2AKKcN5OYakJlbKIqBAAog0UsBcABiABNTQCgACqDQQiaAaAAeQANLfNAmpAAJCAEAz1CBSC3j8v5gKQXgshdC+FiKmAAF9SWDNDCfV5oyIoBSvtMrxIs7H317kslZazPwHB-AVPBuzNJ0ChPLcmtNqZ-Spnw25XQ2D3PxI855jgaVqTfGrX8q5KD-AVYEAukKwUACEiUIqRUwVF6KYaYrmCwDAAQdUhH1Yakl5LKV1CxbaWowsmADNUPJWSAz9mTBhnQX2+CIAcQkKRMNz0ErHFEvYpwl0I2+idJdPZsgHRiCdKxNU1E+nPNdXQWo+RCh0BKOUSo7FfRmEUnxN2ugNxkAwPxPp7pjKtrbcM4Scw62ht9NvEZfTMjMWgOgaSShUA7PQNocQz0g0HLAKS04QA) that I'm hacking on

**Comment by cderv** [comment]
All good thank ! I'll merge this with a minor tweak and we can then look closer to the shiny app and where to publicize it (probably on quarto-web)

**Pull request merged**